External/Plugins/CodeRefactor/Provider/RenamingHelper.cs

```diff
@@ -1,141 +1,166 @@
+                    {
+                        return result;
+                    }
+                }
             }
-            ExecuteFirst();
+            return null;
         }
 
-        static void RenameMember(ClassModel inClass, string name, string newName, bool outputResults)
+        private static Rename RenameMember(ASResult target, string name, string newName, bool outputResults)
         {
-            MemberModel m = inClass.Members.Items.FirstOrDefault(it => it.Name == name);
-            if (m == null) return;
-            ASResult result = new ASResult();
-            ASComplete.FindMember(name, inClass, result, FlagType.Dynamic | FlagType.Function, 0);
-            if (result.Member == null) return;
-            queue.Add(new Rename(result, outputResults, newName));
+            var result = FindGetterSetter(target, name);
+            if (result != null)
+            {
+                return new Rename(result, outputResults, newName);
+            }
+            return null;
         }
-         
-        static void ExecuteFirst()
+
+        private static void ExecuteFirst()
         {
             try
             {
-                currentCommand = queue[0];
-                queue.Remove(currentCommand);
+                currentCommand = queue.Dequeue();
                 currentCommand.OnRefactorComplete += OnRefactorComplete;
                 currentCommand.Execute();
             }
             catch (Exception ex)
             {
                 queue.Clear();
                 currentCommand = null;
+                startState = null;
                 ErrorManager.ShowError(ex);
             }
         }
 
-        static void OnRefactorComplete(object sender, RefactorCompleteEventArgs<IDictionary<string, List<SearchMatch>>> e)
+        private static void OnRefactorComplete(object sender, RefactorCompleteEventArgs<IDictionary<string, List<SearchMatch>>> e)
         {
-            if (queue.Count > 0) ExecuteFirst();
-            else
+            currentCommand.OnRefactorComplete -= OnRefactorComplete;
+            if (queue.Count == 0)
             {
+                //if (currentCommand.OutputResults)
+                //{
+                //    PluginBase.MainForm.CallCommand("PluginCommand", "ResultsPanel.ShowResults");
+                //}
                 if (startState != null) RestoreStartState();
                 currentCommand = null;
                 startState = null;
             }
+            else ExecuteFirst();
+        }
+
+        private static void RestoreStartState()
+        {
+            int pos = startState.CursorPosition;
+            GetOffset(startState.Commands[0], ref pos);
+            GetOffset(startState.Commands[1], ref pos);
+            GetOffset(startState.Commands[2], ref pos);
+            ((ITabbedDocument) PluginBase.MainForm.OpenEditableDocument(startState.FileName)).SciControl.SetSel(pos, pos);
+
+            ASContext.Context.UpdateCurrentFile(true);
```

Comment:
I tried to track the method calls with Go to Definition, but fell into the loop of doom...
I guess I could try to find a cleaner way for the inline rename, like disable all ASContext, ASCompletion and Scintilla functions during rename. 
Because how it currently works is that everything gets reverted back to how it was before inline rename, and then the normal process for Rename is done.